app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/create/SaveChangesDialogFragment.kt

```diff
@@ -0,0 +1,40 @@
+package com.github.alfabravo2013.readyforexams.presentation.create
+
+import android.app.Dialog
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import androidx.navigation.fragment.findNavController
+import com.github.alfabravo2013.readyforexams.R
+import org.koin.android.ext.android.inject
+
+class SaveChangesDialogFragment : DialogFragment() {
+    private val viewModel: SaveChangesDialogViewModel by inject()
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return AlertDialog.Builder(requireContext())
+            .setTitle(getString(R.string.save_changes_dialog_title))
+            .setMessage(getString(R.string.save_changes_dialog_message))
+            .setPositiveButton(getString(R.string.save_changes_dialog_positive)) { _, _ ->
+                viewModel.onPositiveButtonClick()
+                navigateToHomeScreen()
+            }
+            .setNegativeButton(getString(R.string.save_changes_dialog_nagative)) { _, _ ->
+                viewModel.onNegativeButtonClick()
+                navigateToHomeScreen()
```

Comment:
navigateToHomeScreen action should be sent by the ViewModel once you complete the saveChanges and observer by the dialog.